airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/initialsync/MssqlInitialLoadHandler.java

```diff
@@ -0,0 +1,161 @@
+      final Instant emittedAt) {
+    final List<AutoCloseableIterator<AirbyteMessage>> iteratorList = new ArrayList<>();
+    for (final ConfiguredAirbyteStream airbyteStream : catalog.getStreams()) {
+      final AirbyteStream stream = airbyteStream.getStream();
+      final String streamName = stream.getName();
+      final String namespace = stream.getNamespace();
+      // TODO: need to select column according to indexing status of table. may not be primary key
+      final List<String> primaryKeys = stream.getSourceDefinedPrimaryKey().stream().flatMap(pk -> Stream.of(pk.get(0))).toList();
+      final AirbyteStreamNameNamespacePair pair = new AirbyteStreamNameNamespacePair(streamName, namespace);
+      final String fullyQualifiedTableName = DbSourceDiscoverUtil.getFullyQualifiedTableName(namespace, streamName);
+      if (!tableNameToTable.containsKey(fullyQualifiedTableName)) {
+        LOGGER.info("Skipping stream {} because it is not in the source", fullyQualifiedTableName);
+        continue;
+      }
+      if (airbyteStream.getSyncMode().equals(SyncMode.INCREMENTAL)) {
+        // Grab the selected fields to sync
+        final TableInfo<CommonField<JDBCType>> table = tableNameToTable.get(fullyQualifiedTableName);
+        final List<String> selectedDatabaseFields = table.getFields()
+            .stream()
+            .map(CommonField::getName)
+            .filter(CatalogHelpers.getTopLevelFieldNames(airbyteStream)::contains)
+            .toList();
+        primaryKeys.forEach(pk -> {
+          if (!selectedDatabaseFields.contains(pk)) {
+            selectedDatabaseFields.add(0, pk);
+          }
+        });
+
+        final AutoCloseableIterator<JsonNode> queryStream =
+            new MssqlInitialLoadRecordIterator(database, sourceOperations, quoteString, initialLoadStateManager, selectedDatabaseFields, pair, calculateChunkSize(), isCompositePrimaryKey(airbyteStream));
+        final AutoCloseableIterator<AirbyteMessage> recordIterator =
+            getRecordIterator(queryStream, streamName, namespace, emittedAt.toEpochMilli());
+        final AutoCloseableIterator<AirbyteMessage> recordAndMessageIterator = augmentWithState(recordIterator, pair);
+        iteratorList.add(augmentWithLogs(recordAndMessageIterator, pair, streamName));
+      }
+    }
+    return iteratorList;
+  }
+
+  // Transforms the given iterator to create an {@link AirbyteRecordMessage}
+  private AutoCloseableIterator<AirbyteMessage> getRecordIterator(
+      final AutoCloseableIterator<JsonNode> recordIterator,
+      final String streamName,
+      final String namespace,
+      final long emittedAt) {
+    return AutoCloseableIterators.transform(recordIterator, r -> new AirbyteMessage()
+        .withType(Type.RECORD)
+        .withRecord(new AirbyteRecordMessage()
+            .withStream(streamName)
+            .withNamespace(namespace)
+            .withEmittedAt(emittedAt)
+            .withData(r)));
+  }
```

Comment:
nit: perhaps it's just me but if this were inlined I'd find the code more readable. If we're worried about cyclomatic complexity then let's lift L80:99 into a separate function or something.